### DISCUSSION TITLE
How to use a query for time column?

### DISCUSSION THREAD

**Discussion by jakubgs** [apache/superset#20314] [Q&A / Help]
I'm trying to create a burn-up chart of GitHub issues, and I've exported the issues data into a PostgreSQL database:
```
                         Table "public.Issues"
  Column   |           Type           | Collation | Nullable | Default 
-----------+--------------------------+-----------+----------+---------
 id        | character varying(255)   |           | not null | 
 number    | integer                  |           | not null | 
 repo      | character varying(255)   |           | not null | 
 title     | character varying(255)   |           | not null | 
 closed    | boolean                  |           | not null | 
 assignees | integer                  |           | not null | 
 createdAt | timestamp with time zone |           | not null | 
 closedAt  | timestamp with time zone |           |          | 
 updatedAt | timestamp with time zone |           | not null | 
Indexes:
    "Issues_pkey" PRIMARY KEY, btree (id, number
```
I've managed to create a simple burn-up chart by creating two helper views:
```sql
-- Get list of uqnieue dates for all state changes that took place.
CREATE OR REPLACE VIEW "Dates" AS
  SELECT DISTINCT "createdAt"::DATE AS day FROM "Issues"
  UNION
  SELECT DISTINCT "closedAt"::DATE AS day FROM "Issues"
  WHERE "closedAt" IS NOT NULL
  ORDER BY day;

-- Count up all closed and open issues on given date.
SELECT 
   day,
   (SELECT count(*) FROM "Issues" WHERE "createdAt" <= day) AS total,
   (SELECT count(*) FROM "Issues" WHERE "createdAt" <= day AND ("closedAt" IS NULL OR "closedAt" > day)) AS opened,
   (SELECT count(*) FROM "Issues" WHERE "createdAt" <= day AND ("closedAt" IS NOT NULL AND "closedAt" <= day)) AS closed
FROM "Dates";
```
Which can be graphed quite easily due to the `day` column:
![image](https://user-images.githubusercontent.com/2212681/172666216-a9397512-269c-4c99-a00f-dc79408e1c07.png)

But that is not good enough, because we need to be able to filter things on this burn-up chart by various fields, like `repo` for example, or `labels` in the future. But from what I see Superset allows you to use only real columns for the "Time Column":

![image](https://user-images.githubusercontent.com/2212681/172666430-e1624552-94e2-4a4d-9ad1-fe2ef6b4e1f7.png)

So I cannot sure the `UNION` query from above to generate a calculated column and use it for the Time Column in the chart.

Is it possible to chart something like this while retaining the ability to query the issues by other attributes?

**Comment by eran-medan** [comment]
I'm not sure if this helps... but if you have a way to get an audit log of when an issue was opened and closed, you can do the cumulative sum within Superset and use backward fill for empty days. 

<img width="302" alt="image" src="https://github.com/apache/superset/assets/85855778/2569e514-2dfb-41b9-ab4b-37858cd23137">
